internal/controller/clientprofile_controller.go

```diff
@@ -0,0 +1,337 @@
+	}
+	r.cleanUp = r.clientProfile.DeletionTimestamp != nil
+
+	// Validate a pointer to a ceph cluster resource
+	if r.clientProfile.Spec.CephConnectionRef.Name == "" {
+		err := fmt.Errorf("validation error")
+		r.log.Error(err, "Invalid ClientProfile, missing .spec.cephConnectionRef.name")
+		return err
+	}
+
+	// Load the ceph connection
+	r.cephConn.Name = r.clientProfile.Spec.CephConnectionRef.Name
+	r.cephConn.Namespace = r.clientProfile.Namespace
+	if err := r.Get(r.ctx, client.ObjectKeyFromObject(&r.cephConn), &r.cephConn); err != nil {
+		r.log.Error(err, "Failed loading CephConnection")
+		return err
+	}
+
+	// Ensure the CephConnection has an owner reference (not controller reference)
+	// for the current reconciled ClientProfile
+	cephConnHasOwnerRef := false
+	for i := range r.cephConn.OwnerReferences {
+		ownerRef := &r.cephConn.OwnerReferences[i]
+		if ownerRef.UID == r.clientProfile.UID {
+			cephConnHasOwnerRef = true
+			break
+		}
+	}
+	if !cephConnHasOwnerRef {
+		if err := ctrlutil.SetOwnerReference(&r.clientProfile, &r.cephConn, r.Scheme); err != nil {
+			r.log.Error(err, "Failed adding an owner reference on CephConnection")
+			return err
+		}
+		r.log.Info("Owner reference missing on CephConnection, adding")
+		if err := r.Update(r.ctx, &r.cephConn); err != nil {
+			r.log.Error(err, "Failed adding an owner reference to CephConnection")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *ClientProfileReconcile) reconcileCephConnection() error {
+	log := r.log.WithValues("cephConnectionName", r.cephConn.Name)
+	log.Info("Reconciling CephConnection")
+
+	if needsUpdate, err := utils.ToggleOwnerReference(
+		!r.cleanUp,
+		&r.clientProfile,
+		&r.cephConn,
+		r.Scheme,
+	); err != nil {
+		r.log.Error(err, "Failed to toggle owner reference on CephConnection")
+		return err
+	} else if needsUpdate {
+		if err := r.Update(r.ctx, &r.cephConn); err != nil {
+			r.log.Error(err, "Failed to update CephConnection")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *ClientProfileReconcile) reconcileCephCsiClusterInfo() error {
+	csiConfigMap := corev1.ConfigMap{}
+	csiConfigMap.Name = utils.CsiConfigVolume.Name
+	csiConfigMap.Namespace = r.clientProfile.Namespace
+
+	log := r.log.WithValues("csiConfigMapName", csiConfigMap.Name)
+	log.Info("Reconciling Ceph CSI Cluster Info")
+
+	// Using a lock to serialized the updating of the config map.
+	// Although the code will run perfetcly fine without the lock, there will be a higher
+	// chance to fail on the create/update operation because another concurrent reconcile loop
+	// updated the config map which will result in stale representation and an update failure.
+	// The locking strategy will sync all update to the shared config file and will prevent such
+	// potential issues without a big impact on preformace as a whole
+	configMapUpdateLock.Lock()
+	defer configMapUpdateLock.Unlock()
+
+	_, err := ctrlutil.CreateOrUpdate(r.ctx, r.Client, &csiConfigMap, func() error {
```

Comment:
so the configmap creation happen only after profile is created, were any changes already made that driver/operatorconfig doesn't look for this before they deploy CSI?

Reply:
This is something the was already discussed and I will provide these changes in a future PR

Reply:
Ack, resolve.